core/domain/user_query_jobs_one_off.py

```diff
@@ -114,6 +120,35 @@ def _is_maximum_exp_edited_query_satisfied(
             len(user_contributions.edited_exploration_ids) <
             query_model.edited_fewer_than_n_exps)
 
+    @staticmethod
+    def _is_created_collection_query_satisfied(
+            user_settings_model, _):
+        """Determines whether a user has created collections."""
+        user_id = user_settings_model.id
+        collection = collection_models.CollectionRightsModel.query(
+            collection_models.CollectionRightsModel.owner_ids == user_id
+        ).get()
+        return collection is not None
+
+    @staticmethod
+    def _is_used_logic_proof_interaction_query_satisfied(
+            user_settings_model, _):
+        """Determines whether a user has used logic proof
+        interaction in any of the explorations created by the user.
+        """
+        user_id = user_settings_model.id
+        user_contributions = user_models.UserContributionsModel.get(user_id)
+        exploration_ids = user_contributions.created_exploration_ids
+        exploration_instances = (
+            datastore_services.fetch_multiple_entities_by_ids_and_models(
+                [('ExplorationModel', exploration_ids)]))[0]
+        for item in exploration_instances:
+            exploration = exp_fetchers.get_exploration_from_model(item)
+            for _, state in exploration.states.items():
+                if state.interaction.id == 'LogicProof':
```

Comment:
Is the plan to remove the LogicProof interaction entirely? Will we be deleting the interactions that point to it?

Reply:
We have just now started and are working on how to remove the LogicProof interaction. We plan to remove LogicProof interaction entirely as it slows down the website. I think we will be deleting or modifying those interactions.